apis/stardust/types/index.d.ts

```diff
@@ -261,6 +261,19 @@ declare namespace stardust {
         anything?: object;
     }
 
+    interface Navigation {
+        getUrlForSheet?: (sheetId: string) => string;
+        goToSheet?: (sheetId: string, chartId?: string) => void;
+        nextSheet?: () => boolean;
+        prevSheet?: () => boolean;
+        goToStory?: (storyId: string) => void;
+        getCurrentSheetId?: () => string | boolean;
+        getCurrentStoryId?: () => string | boolean;
+        refreshDynamicViews?: () => void;
+        getOdagLinks?: (app: qix.Doc) => unknown[];
+        openOdagPopup?: (app: qix.Doc, odagLinkId: string, element: HTMLButtonElement) => void;

```

Comment:
Where did all this come from?

Reply:
This file is auto generated from the JSdoc comments, so those should be updated first. Also this isn't the Navigation API from client, its most often just a subset.

Reply:
I wrote the types myself according to https://github.com/qlik-trial/sense-client/blob/main/web/assets/objects/extension/supernova-wrapper/navigation.js

Reply:
I just put optional object for navigation in context.